unix/entry.c

```diff
@@ -73,6 +92,9 @@ static uiEntry *finishNewEntry(GtkWidget *w, const gchar *signal)
 	e->onChangedSignal = g_signal_connect(e->widget, signal, G_CALLBACK(onChanged), e);
 	uiEntryOnChanged(e, defaultOnChanged, NULL);
 
+	g_signal_connect(e->widget, "activate", G_CALLBACK(onFinished), e);
```

Comment:
Would `editing-done` (in GtkCellEditable) be a better signal here? The docs seem to imply they are connected, but...

Note to self: TODO investigate the phrasing "it is also commonly used by applications to intercept activation of entries" in the docs.

Reply:
I'd discounted `editing-done` because `GtkCellEditable` is designed for entry widgets embedded in tables rather than stand-alone. But the docs do rather imply that stock `GtkEntry` always supports the `GtkCellEditable` interface.
So I had another go at it, but no dice. Just couldn't see any `editing-done` signals triggering. I might have been doing it all wrong, but a internet search seems strangely silent on the whole subject.
Like you said, time to dig through the Gtk source code.

I see the "it is also commonly used by applications to intercept activation of entries" line as acknowledging that the `activate` signal is used for this purpose by a lot of apps, even if it wasn't originally intended for this use. And, given that, I reckon `activate` is a sound enough method - it works and is unlikely to be removed any time soon.

Reply:
Ahh - I've just been poking through the Gtk+ source code.
Calling `gtk_cell_editable_start_editing ()` on `GtkEntry` will cause it to wire up it's `activate` signal to a callback that emits both the `editing-done` and `remove-widget` `GtkCellEditable` signals.
(see the `GtkCellEditable` implementation for `GtkEntry`: https://git.gnome.org/browse/gtk+/tree/gtk/gtkentry.c#n4587 )

So I think we're better off sticking with `activate`.